src/catalog/src/expr_cache.rs

```diff
@@ -229,10 +215,16 @@ impl ExpressionCache {
                                 continue;
                             }
                         };
-                        // Remove dropped IDs and global expressions that were cached with different
-                        // features.
+                        // Remove dropped IDs and expressions that rely on dropped indexes.
+                        let index_dependencies: BTreeSet<_> = expressions
+                            .global_mir
+                            .index_imports
+                            .keys()
+                            .chain(expressions.physical_plan.index_imports.keys())
+                            .cloned()
+                            .collect();
                         if !current_ids.contains(&key.id)
-                            || expressions.optimizer_features != *optimizer_features
+                            || !index_dependencies.is_subset(current_ids)
```

Comment:
@ggevay does this make sense in terms of the calls to `index_imports`? 

This is called during startup. I have a set of all the items in the catalog, `current_ids`. I'm trying to remove any expression from the cache that may rely on an index that no longer exists, i.e. doesn't exist in `current_ids`. Should I also be looking in other places inside of `expressions`?

Reply:
Looking at `index_imports` should be enough: A dataflow can use an index only if the index is imported.

(Note that, currently, the `index_imports` of MIR and LIR plan are the same, so strictly speaking it would be enough to look at only one of them. But I think it's good practice if you check both here (as it is currently done in your PR), so that your code is future-proof against possible changes where the MIR and LIR `index_imports` maybe drift apart.)